apps/browser/qml/pages/components/SecureAction.qml

```diff
@@ -0,0 +1,33 @@
+/****************************************************************************
+**
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.0
+import com.jolla.settings.system 1.0
+import org.nemomobile.devicelock 1.0
+
+DeviceLockQuery {
+    id: deviceLockQuery
+    property bool requireAuthentication: true
+    property string message
+
+    function perform(resolve) {
+        if (requireAuthentication && _availableMethods !== Authenticator.NoAuthentication) {
```

Comment:
There are few sources possibly causing this misunderstanding but this should be more like SecurityCodeSettings.set instead of checking available method. To clarify, available methods are for device lock automatic locking setting. More specificly, Authenticator.NoAuthentication is available method when automatic locking is set to "Not in use". That still means that device lock code is set if SecurityCodeSettings.set.

Reply:
Thanks for clarifying this, it's a really good point. I've updated the code to use `SecurityCodeSettings.set` instead.

Reply:
You shouldn't need to check at all, there is some sort of option to force a prompt but the default behavior is to authenticate if no code is set.

Reply:
Sorry, the requestPermission() function on Authenticator will still display a confirmation prompt if no security methods are enabled, it's authenticating a challenge code that will auto grant unless told otherwise.

Checking the availableMethods should be correct though. Those are the available methods for authentication, unlocking the device is done through the DeviceLock object and available methods for that aren't exposed through the API because that and the device lock state are managed internally in the daemon. If Authenticator.availableMethods changed based on the automatic locking setting, that would be a bug.

Reply:
I've switched to using the `Authenticator` code you suggested [below](https://github.com/sailfishos/sailfish-browser/pull/838#discussion_r647931330) @adenexter, which uses `availableMethods`.